### PR TITLE
Add Workflows list endpoint

### PR DESCRIPTION
**Pull request by vegaro**
## Summary

- Adds `WorkflowSummary` and `WorkflowsListResponse` models (`id`, `display_name`, `offering_id`, `prefetch`) with kotlinx.serialization and `ignoreUnknownKeys` for forward compatibility
- Adds `WorkflowJsonParser.parseWorkflowsListResponse()` 
- Adds `Endpoint.GetWorkflows` and `Backend.getWorkflows()` with background-aware callback deduplication (same pattern as the existing `getWorkflow()` implementation)

Part 1 of 2 — Part 2: #3508

---

> [!NOTE]
> **Low Risk**
> Additive networking and parsing alongside existing workflow detail APIs, with broad test coverage and no changes to purchase or auth flows.
> 
> **Overview**
> Adds client support for **listing subscriber workflows** via `GET /v1/subscribers/{userId}/workflows`, including an optional `type` query filter.
> 
> New kotlinx-serialization models `WorkflowSummary` and `WorkflowsListResponse` (with `parseWorkflowsListResponse`) describe list entries (`id`, `display_name`, `offering_id`, `prefetch`). `Endpoint.GetWorkflows` is wired for signature verification like the single-workflow endpoint, and **`Backend.getWorkflows`** performs the request with the same background-aware callback deduplication pattern as `getWorkflow`. Tests cover path encoding, deserialization edge cases, HTTP errors, and concurrent call coalescing.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b27d7cdbb89f44573d87272a7cbdd37f19a4d8db. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: purchases/src/main/kotlin/com/revenuecat/purchases/common/networking/Endpoint.kt [hunk 1]
```diff
@@ -35,6 +35,13 @@ internal sealed class Endpoint(
         override fun getPath(useFallback: Boolean) =
             pathTemplate.format(Uri.encode(userId), Uri.encode(workflowId))
     }
+    data class GetWorkflows(val userId: String) : Endpoint(
+        "/v1/subscribers/%s/workflows",
+        "get_workflows",
+    ) {
+        override fun getPath(useFallback: Boolean) =
+            pathTemplate.format(Uri.encode(userId))
+    }
```
[the remark this answers is not in the document]
Comment (facumenzella) [reply purchases/src/main/kotlin/com/revenuecat/purchases/common/networking/Endpoint.kt:46]: +1

File: purchases/src/test/java/com/revenuecat/purchases/common/backend/BackendWorkflowsTest.kt [hunk 2]
```diff
@@ -176,6 +193,108 @@ class BackendWorkflowsTest {
...
+                baseURL = mockBaseURL,
+                endpoint = Endpoint.GetWorkflows(appUserId),
+                body = null,
+                postFieldsToSign = null,
+                requestHeaders = defaultAuthHeaders,
+                fallbackBaseURLs = emptyList(),
+            )
+        } answers {
+            Thread.sleep(200)
+            httpResult(RCHTTPStatusCodes.SUCCESS, listJson)
```
Comment (facumenzella) [inline purchases/src/test/java/com/revenuecat/purchases/common/backend/BackendWorkflowsTest.kt]: I guess this sleep is standard? 🤔

**Review by facumenzella — APPROVED** [review]
looks solid

File: purchases/src/test/java/com/revenuecat/purchases/common/backend/BackendWorkflowsTest.kt (hunk 2 above)
Comment (vegaro) [reply purchases/src/test/java/com/revenuecat/purchases/common/backend/BackendWorkflowsTest.kt]: I can try a different way
